google/cloud/bigquery/job.py

```diff
@@ -1042,7 +1042,7 @@ def autodetect(self, value):
 
     @property
     def clustering_fields(self):
-        """Union[List[str], None]: Fields defining clustering for the table
+        """Optional[Union[List[str], None]]: Fields defining clustering for the table
```

Comment:
```suggestion
        """Optional[List[str]]: Fields defining clustering for the table
```
`Union[X, None]` is actually a [longer version](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#optional-types-and-the-none-type) of `Optional[X]`.